lib/gosquared/test.js

```diff
@@ -120,22 +120,98 @@ describe('GoSquared', function(){
         analytics.stub(window, '_gs');
       });
 
+      var created = new Date();
+
       it('should set an id', function(){
         analytics.identify('id');
         analytics.called(window._gs, 'identify', 'id');
       });
 
-      it('should set traits', function(){
-        analytics.identify({ trait: true });
-        analytics.called(window._gs, 'properties', { trait: true });
+      it('should alias traits', function(){
+        analytics.identify({
+          firstName: 'first_name',
+          lastName: 'last_name',
+          createdAt: created,
+          title: 'company_position',
+          industry: 'company_industry'
+        });
+        analytics.called(window._gs, 'properties', {
+          first_name: 'first_name',
+          last_name: 'last_name',
+          created_at: created,
+          company_position: 'company_position',
+          company_industry: 'company_industry'
+        });
+      });
+
+      it('should keep special traits in root object', function(){
+        analytics.identify({
+          id: 'id',
+          email: 'email',
+          firstName: 'first_name',
+          lastName: 'last_name',
+          username: 'username',
+          description: 'description',
+          avatar: 'avatar',
+          phone: 'phone',
+          createdAt: created,
+          company_name: 'company_name',
+          company_size: 'company_size',
+          title: 'company_position',
+          industry: 'company_industry'
+        });
+        analytics.called(window._gs, 'properties', {
+          id: 'id',
+          email: 'email',
+          first_name: 'first_name',
+          last_name: 'last_name',
+          username: 'username',
+          description: 'description',
+          avatar: 'avatar',
+          phone: 'phone',
+          created_at: created,
+          company_name: 'company_name',
+          company_size: 'company_size',
+          company_position: 'company_position',
+          company_industry: 'company_industry'
+        });
+      });
+
+      it('should move non-special traits to custom object', function(){
+        analytics.identify({
+          trait: true,
+          weirdTraits: 'can go anywhere'
+        });
+        analytics.called(window._gs, 'properties', {
+          custom: {
+            trait: true,
+            weirdTraits: 'can go anywhere'
+          }
+        });
+      });
+
+      it('should keep custom traits and ignore non-special properties', function(){
+        analytics.identify({
+          trait: true,
+          weirdTraits: 'cant go anywhere',
+          custom: {
+            iWantThese: 'traits'
+          }
+        });
+        analytics.called(window._gs, 'properties', {
+          custom: {
+            iWantThese: 'traits'
```

Comment:
hmmmm, yeah i think we want to avoid provider-specific things like this `custom` behavior. @simontabor can we do without that `custom` special behavior? not sure what the use case for a customer is there.